pkg/mtbroker/filter/filter_handler.go

```diff
@@ -218,8 +219,15 @@ func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers
 	statusCode, err := writeResponse(ctx, writer, response, ttl, span)
 	if err != nil {
 		h.logger.Error("failed to write response", zap.Error(err))
+		// Ok, so writeResponse will return the HttpStatus of the function. That may have
+		// succeeded (200), but it may have returned a malformed event, so if the
+		// function succeeded, convert this to an StatusBadGateway instead to indicate
+		// error. Note that we could just use StatusInternalServerError, but to distinguish
+		// between the two failure cases, we use a different code here.
+		if statusCode == 200 {
```

Comment:
I threw up in my mouth a little. 

I will look at fixing this upstream a little.